cmd/disk-cache.go

```diff
@@ -303,7 +303,7 @@ func (c cacheObjects) GetObject(ctx context.Context, bucket, object string, star
 		}
 		dcache.Delete(ctx, bucket, object)
 	}
-	if startOffset != 0 || length != objInfo.Size {
+	if startOffset != 0 || (length > 0 && length != objInfo.Size) {
```

Comment:
I think it is just easier if it is this way:
```
if startOffset != 0 || length == -1 || length != objInfo.Size {
```

Reply:
Actually we need to cache when length == -1 and startOffset == 0

Reply:
> Actually we need to cache when length == -1 and startOffset == 0

yes right